abdm_integrator/hiu/views/consents.py

```diff
@@ -0,0 +1,225 @@
+        return consent_request
+
+
+class GatewayConsentRequestOnInit(HIUGatewayBaseView):
+    def post(self, request, format=None):
+        GatewayConsentRequestOnInitSerializer(data=request.data).is_valid(raise_exception=True)
+        self.process_request(request.data)
+        return Response(status=HTTP_202_ACCEPTED)
+
+    def process_request(self, request_data):
+        consent_request = ConsentRequest.objects.get(
+            gateway_request_id=request_data['resp']['requestId']
+        )
+        if request_data.get('consentRequest'):
+            consent_request.consent_request_id = request_data['consentRequest']['id']
+            consent_request.status = ConsentStatus.REQUESTED
+        elif request_data.get('error'):
+            consent_request.status = ConsentStatus.ERROR
+            consent_request.error = request_data['error']
+        consent_request.save()
+
+
+class GatewayConsentRequestNotify(HIUGatewayBaseView):
+
+    def post(self, request, format=None):
+        GatewayConsentRequestNotifySerializer(data=request.data).is_valid(raise_exception=True)
+        # TODO Run as a background task in celery
+        GatewayConsentRequestNotifyProcessor(request.data).process_request()
+        return Response(status=HTTP_202_ACCEPTED)
+
+
+class GatewayConsentRequestNotifyProcessor:
+
+    def __init__(self, request_data):
+        self.request_data = request_data
+
+    @transaction.atomic
+    def process_request(self):
+        consent_status = self.request_data['notification']['status']
+        if consent_status == ConsentStatus.REVOKED:
+            return self.handle_revoked()
+        consent_request = ConsentRequest.objects.get(
+            consent_request_id=self.request_data['notification']['consentRequestId']
+        )
+        consent_request.update_status(consent_status)
+        if consent_status == ConsentStatus.GRANTED:
+            self.handle_granted(consent_request)
+        elif consent_status == ConsentStatus.EXPIRED:
+            self.handle_expired(consent_request)
+
+    def handle_granted(self, consent_request):
+        for artefact in self.request_data['notification']['consentArtefacts']:
+            consent_artefact = ConsentArtefact(artefact_id=artefact['id'], consent_request=consent_request)
+            self.artefact_fetch(consent_artefact)
+
+    def handle_expired(self, consent_request):
+        artefact_ids = []
+        for artefact in consent_request.artefacts.all():
+            artefact_ids.append(artefact.artefact_id)
+            artefact.delete()
+        self.gateway_consents_on_notify(artefact_ids)
+
+    def handle_revoked(self):
+        artefact_ids = [artefact['id'] for artefact in self.request_data['notification']['consentArtefacts']]
+        consent_request = ConsentRequest.objects.get(artefacts__artefact_id=artefact_ids[0])
+        for artefact_id in artefact_ids:
+            ConsentArtefact.objects.get(artefact_id=artefact_id).delete()
```

Comment:
```suggestion
        ConsentArtefact.objects.filter(artefact_id__in=artefact_ids).delete()
```

Reply:
Addressed in e593a740cac779eb4c82df121e8f859458033f57